patches/api/0490-Add-enchantWithLevels-with-tag-specification.patch

```diff
@@ -0,0 +1,32 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: masmc05 <masmc05@gmail.com>
+Date: Wed, 25 Sep 2024 16:26:04 +0300
+Subject: [PATCH] Add enchantWithLevels with tag specification
+
+
+diff --git a/src/main/java/org/bukkit/inventory/ItemFactory.java b/src/main/java/org/bukkit/inventory/ItemFactory.java
+index bd0e55562f1cabef3078573182e0cf9fbc844585..2b216b1ee98ef153bdcc757f0f241a59bb4f7f75 100644
+--- a/src/main/java/org/bukkit/inventory/ItemFactory.java
++++ b/src/main/java/org/bukkit/inventory/ItemFactory.java
+@@ -338,4 +338,21 @@ public interface ItemFactory {
+      */
+     @NotNull ItemStack enchantWithLevels(@NotNull ItemStack itemStack, @org.jetbrains.annotations.Range(from = 1, to = 30) int levels, boolean allowTreasure, @NotNull java.util.Random random);
+     // Paper end - enchantWithLevels API
++    // Paper start - enchantWithLevels with tag specification
++    /**
++     * Randomly enchants a copy of the provided {@link ItemStack} using the given experience levels.
++     *
++     * <p>If the provided ItemStack is already enchanted, the existing enchants will be removed before enchanting.</p>
++     *
++     * <p>Levels must be in range {@code [1, 30]}.</p>
++     *
++     * @param itemStack ItemStack to enchant
++     * @param levels levels to use for enchanting
++     * @param tagKey tag key which will be used to determine the enchantments to apply (e.g. {@link io.papermc.paper.registry.keys.tags.EnchantmentTagKeys#IN_ENCHANTING_TABLE})
++     * @param random {@link java.util.Random} instance to use for enchanting
++     * @return enchanted copy of the provided ItemStack
++     * @throws IllegalArgumentException on bad arguments
++     */
++    @NotNull ItemStack enchantWithLevels(@NotNull ItemStack itemStack, @org.jetbrains.annotations.Range(from = 1, to = 30) int levels, @NotNull io.papermc.paper.registry.tag.TagKey<Enchantment> tagKey, @NotNull java.util.Random random);
```

Comment:
It would probably be better to make the parameter type, `RegistryKeySet<Enchantment>`, that way it could be a tag **or** just a specific set enchants a plugin might want to have available.